### PR TITLE
Add a Dockerfile for felix

### PR DESCRIPTION
**Pull request by tomdee**
- Build felix (and deps) inside the container
- Resulting container is ~60MB (mostly python and deps)

### CODE REVIEW

**Comment by alexwlchan** [comment]
retest this please

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# The final container is slightly bloated by having the source code
+# But it's only a couple of MB (kept down using .dockerignore)
+ADD . /code
+WORKDIR /code
+
+RUN apk -U add python py-setuptools libffi ip6tables ipset iputils && \
+    apk add --virtual temp python-dev libffi-dev py-pip alpine-sdk && \
+    pip install -e . && \
+    apk del temp && rm -rf /var/cache/apk/*
+RUN mkdir /etc/calico && echo -e "[global]\nMetadataAddr = None\nLogFilePath = None\nLogSeverityFile = None" >/etc/calico/felix.cfg
```
Comment (fasaxc) [inline Dockerfile]: Nicer to put this in a dedicated file?

**Comment by fasaxc** [comment]
@tomdee can you give me a bit more context on why you're adding this?  Is it just to scratch an itch or are you looking to use it with calico-docker, for example?

File: Dockerfile (hunk 1 above)
Comment (tomdee) [inline Dockerfile]: I agree but I wasn't sure where to put it so that it doesn't confuse people. Any suggestions?

**Comment by tomdee** [comment]
Hopefully we'll be splitting up the big calico-node container into smaller single-process-containers. (Or at least having that as an option). But that's not agreed yet, so for the moment this is just scratching an itch. Seems like it might be useful for people in general to get a simple way to "install" felix, as a cross OS alternative to debs or RPMs.

File: Dockerfile (hunk 1 above)
Comment (fasaxc) [inline Dockerfile]: How about putting it in "<repo>/docker/felix.cfg"?

The Dockerfile and .dockerignore have to be at the top-level, right?

**Comment by fasaxc** [comment]
Sounds good.  We could let this go in as-is but would you mind putting a note in the README or the docs explaining how to use it?  I expect it needs `--net=host`, for example?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# edge is requried to get the fix for https://bugs.alpinelinux.org/issues/4451
```
Comment (alexwlchan) [inline Dockerfile]: /s/requried/required

**Comment by tomdee** [comment]
@fasaxc Sorry I've left this so long. I would now like to use this to create a rkt pod. rkt 1.0 is being released next week so I need to move really quickly. I've made the requested changes and merged it so I can start setting up dockerhub. If you need me to make any more changes then let me know.

**Pull request merged**